### PR TITLE
add keras_tuner tensorboard guide

### LINKED ISSUE #492: Add a Keras Tuner guide, visualize tuning process with TensorBoard
Assign to @haifeng-jin

### PR DESCRIPTION
**Pull request by haifeng-jin**
Since we are improving the visual tuning experience,
this guide is a temporary one.
Will have a more comprehensive guide after more features are added.
Probably will add a video guide then.

resolve #492

### CODE REVIEW

File: guides/keras_tuner/visualize_tuning.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+(multi-layer perceptron or convolutional neural network), the number of layers,
+the number of units or filters, whether to use dropout.
+"""
+
+
+def build_model(hp):
+    num_classes = 10
+    input_shape = (28, 28, 1)
+    model = keras.Sequential()
+    model.add(keras.layers.InputLayer(input_shape=input_shape))
```
Comment (fchollet) [inline guides/keras_tuner/visualize_tuning.py:66]: Prefer using the Functional API here.

File: guides/keras_tuner/visualize_tuning.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    model_type = hp.Choice("model_type", ["mlp", "cnn"])
+    if model_type == "mlp":
+        model.add(layers.Flatten())
+        # Number of layers of the MLP is a hyperparameter.
+        for i in range(hp.Int("mlp_layers", 1, 3)):
+            # Number of units of each layer are
+            # different hyperparameters with different names.
+            model.add(
+                layers.Dense(
+                    units=hp.Int("units_{i}".format(i=i), 32, 128, step=32),
```
Comment (fchollet) [inline guides/keras_tuner/visualize_tuning.py:77]: You can save quite a few characters by using f strings, `f"units_{i}"`

File: guides/keras_tuner/visualize_tuning.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    if hp.Boolean("dropout"):
+        model.add(layers.Dropout(0.5))
+
+    model.add(layers.Dense(units=num_classes, activation="softmax"))
+
+    # Compile the model.
+    model.compile(
+        loss="sparse_categorical_crossentropy",
+        metrics=["accuracy"],
+        optimizer=hp.Choice("optimizer", ["adam", "adadelta", "sgd"]),
```
Comment (fchollet) [inline guides/keras_tuner/visualize_tuning.py:104]: `adadelta` with default parameters is generally a bad choice (it needs a lr closer to 1 but by default uses 1e-3). You already have lots of hps in this example so tuning the optimizer isn't super necessary

File: guides/keras_tuner/visualize_tuning.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    )
+    return model
+
+
+"""
+We can do a quick test of the models to check if it build successfully for both
+CNN and MLP.
+"""
+
+import keras_tuner as kt
```
Comment (fchollet) [inline guides/keras_tuner/visualize_tuning.py:114]: Move it to the top

File: guides/keras_tuner/visualize_tuning.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+"""
+Start the search by calling `tuner.search(...)`. To use TensorBoard, we need
+to pass a `keras.callbacks.TensorBoard` instance to the callbacks.
+"""
+
+tuner.search(
+    x_train,
+    y_train,
+    validation_data=(x_test, y_test),
```
Comment (fchollet) [inline guides/keras_tuner/visualize_tuning.py:155]: Never use the test set as validation data during tuning. In this example, you can use a `validation_split` and finally test your best model(s) on the test set at the end

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! Please only add the generated files after the PR is approved (otherwise they have to be regenerated)

**Review by fchollet — COMMENTED** [review]
Looking great! It may be that an extensive set of screenshots (with comments describing them) would work better than an embedded video.

**Comment by haifeng-jin** [comment]
To the ease of viewing the website with images, I've generated the md file.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
